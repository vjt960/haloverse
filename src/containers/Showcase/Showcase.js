import React, { Component } from 'react';
import { connect } from 'react-redux';

export class Showcase extends Component {
  constructor(props) {
    super();
    this.state = {
      mapName: '',
      mapURL: '',
      description: '',
      index: Math.floor(Math.random() * Math.floor(props.maps.length))
    };
  }

  componentDidMount() {
    this.setState({
      mapName: this.props.maps[this.state.index].name,
      mapURL: this.props.maps[this.state.index].imageUrl,
      description: this.props.maps[this.state.index].description
    });
  }

  render() {
    return (
      <section
        className="main_showcase"
        style={{
          border: '1px solid magenta',
          height: '70vh',
          width: '100%',
          backgroundImage: 'url(' + this.state.mapURL + ')',
          backgroundSize: 'cover',
          backgroundPosition: '50% 50%',
          backgroundRepeat: 'no-repeat',
          objectFit: 'cover'
        }}
      >
        <p className="main_showcase--map-title">{this.state.mapName}</p>
        <p className="main_showcase--map-description">
          {this.state.description}
        </p>
      </section>
    );
  }
}

const mapStateToProps = state => ({
  isLoading: state.isLoading,
  maps: state.maps
});

export default connect(mapStateToProps);
