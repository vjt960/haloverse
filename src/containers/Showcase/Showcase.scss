.main_showcase {
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
  align-items: flex-start;
  padding: 20px;
  color: hsl(0, 0%, 100%);
}

.main_showcase--map-title {
  font-size: 1.75em;
}

.main_showcase--map-description {
  font-size: 1.35em;
}