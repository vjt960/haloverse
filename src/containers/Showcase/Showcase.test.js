import React from 'react';
import { shallow } from 'enzyme';
import { Showcase, mapStateToProps } from './Showcase';

describe('Showcase', () => {
  describe('Showcase Component', () => {
    let wrapper;

    beforeEach(() => {
      const props = { maps: [1, 2, 3], isLoading: false };
      wrapper = shallow(<Showcase {...props} />);
    });

    it('should match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('mapStateToProps', () => {
    it('returns an object with isLoading and maps from state', () => {
      const mockState = {
        isLoading: false,
        maps: [1, 2, 3],
        extraProp: 'fill',
        uselessProp: 'fill'
      };
      const expected = { isLoading: false, maps: [1, 2, 3] };
      const mappedProps = mapStateToProps(mockState);
      expect(mappedProps).toEqual(expected);
    });
  });
});
