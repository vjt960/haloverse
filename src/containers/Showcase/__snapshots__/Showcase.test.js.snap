// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Showcase Showcase Component should match snapshot 1`] = `ShallowWrapper {}`;
